### PR TITLE
More interop cleanup

### PR DESCRIPTION
**Pull request by JeremyKuhne**
Some more interop cleanup in preparation of moving other Win32 interop to a shared model with System.Drawing.

- Add System and System.Collections.Generic to global usings for System.Drawing
- Move Libraries and NullHandle to Core
- Move scopes to Gdi32 namespace and out of PInvoke (necessary for now as we now have multiple PInvoke classes)
- Move COM related stuff to Com namespace

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/winforms/pull/10613)

### CODE REVIEW

File: src/System.Windows.Forms.Primitives/src/Windows/Win32/ComHelpers.cs [hunk 1]
```diff
@@ -18,27 +18,26 @@ internal static unsafe partial class ComHelpers
     /// </summary>
     internal static bool BuiltInComSupported { get; }
         = AppContext.TryGetSwitch("System.Runtime.InteropServices.BuiltInComInterop.IsSupported", out bool supported)
-            ? supported
-            : true;
+            && supported;
```
Comment (lonitra) [inline src/System.Windows.Forms.Primitives/src/Windows/Win32/ComHelpers.cs:21]: Should this be:
```c#
!AppContext.TryGetSwitch("System.Runtime.InteropServices.BuiltInComInterop.IsSupported", out bool supported)
            || supported;
```

File: src/System.Windows.Forms/src/System/Windows/Forms/NativeWindow.cs [hunk 2]
```diff
@@ -500,7 +499,7 @@ public virtual unsafe void CreateHandle(CreateParams cp)
                     // based on the window properties and the behavior of the thread. For additional information, please refer to
                     // https://microsoft.visualstudio.com/OS/_git/os.2020?path=/clientcore/windows/Core/ntuser/kernel/windows/createw.cxx&version=GBofficial/main&line=881&lineEnd=882&lineStartColumn=1&lineEndColumn=1&lineStyle=plain&_a=contents
                     DPI_AWARENESS_CONTEXT controlHandleDpiContext = PInvoke.GetWindowDpiAwarenessContext(HWND);
-                    Debug.Assert(PInvoke.AreDpiAwarenessContextsEqualInternal(DpiAwarenessContext, controlHandleDpiContext),
+                    Debug.Assert(!DpiAwarenessContext.IsEquivalent(controlHandleDpiContext),
```
Comment (lonitra) [inline src/System.Windows.Forms/src/System/Windows/Forms/NativeWindow.cs:502]: Why did we do `!` here?

File: src/System.Windows.Forms/src/System/Windows/Forms/Rendering/DCMapping.cs [hunk 3]
```diff
@@ -59,49 +58,49 @@ public unsafe DCMapping(HDC hdc, Rectangle bounds)
...
+                selectResult is GDI_REGION_TYPE.SIMPLEREGION or GDI_REGION_TYPE.NULLREGION,
                 "SIMPLEREGION or NULLLREGION expected.");
         }
         catch (Exception ex) when (!ex.IsCriticalException())
         {
         }
     }
 
-    public void Dispose()
+    public readonly void Dispose()
```
Comment (lonitra) [inline src/System.Windows.Forms/src/System/Windows/Forms/Rendering/DCMapping.cs:103]: Is adding this `readonly` needed is the struct is already `readonly`?

Comment (JeremyKuhne) [reply src/System.Windows.Forms/src/System/Windows/Forms/Rendering/DCMapping.cs:103]: Maybe not. I think I auto-fixed that before the struct.

File: src/System.Windows.Forms/src/System/Windows/Forms/NativeWindow.cs (hunk 2 above)
Comment (JeremyKuhne) [reply src/System.Windows.Forms/src/System/Windows/Forms/NativeWindow.cs:502]: Whoopsie

**Review by lonitra — APPROVED** [review]
🚀

**Pull request merged**